pkg/sync/rpc/server.go

```diff
@@ -1505,3 +1529,137 @@ func (b *BackupList) Delete(backupID string) error {
 	}
 	return fmt.Errorf("backup not found %v", backupID)
 }
+
+func (s *SyncAgentServer) SnapshotHash(ctx context.Context, req *ptypes.SnapshotHashRequest) (*empty.Empty, error) {
+	if len(req.SnapshotNames) == 0 {
+		return nil, fmt.Errorf("snapshot name(s) is required")
+	}
+
+	status := hash.DoSnapshotHash(req.SnapshotNames, req.Rehash, req.ConcurrentLimit)
+
+	if err := s.SnapshotHashList.Delete(req.SnapshotNames[0]); err != nil {
+		return nil, errors.Wrapf(err, "failed to delete snapshot %v for from list", req.SnapshotNames[0])
+	}
+
+	if err := s.SnapshotHashList.Add(req.SnapshotNames[0], status); err != nil {
+		return nil, errors.Wrapf(err, "failed to add snapshot %v for hashing", req.SnapshotNames[0])
+	}
```

Comment:
This part looks a little bit confusing with the first glance. Can we add a comment that describes how the syne agent server updates `SnapshotHashList`?
1. Typically, for this kind of operation, we will add the lock, update the list, use a defer function (to finish the handling) then start the actual operation.
2. Why does it handle the 1st entry of the list only?